src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int MAX_CAPACITY = 1 << 30;
+    private static final int SIZE_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int currentCapacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+        this.currentCapacity = INITIAL_CAPACITY;
+        this.threshold = (int) (INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = defineCell(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            table[index].value = value;
+            size++;
+        } else {
+            Node<K, V> currNode = table[index];
+            while (currNode != null) {
+                if (currNode.key == key || Objects.equals(key, currNode.key)) {
```

Comment:
If objects are equal, their hashcodes are equal too
If you compare keys, why do you need to compare hashs too?